packages/components/src/utils/element-rect.ts

```diff
@@ -152,28 +150,44 @@ export const NULL_ELEMENT_OFFSET_RECT = {
 
 /**
  * Returns the position and dimensions of an element, relative to its offset
- * parent. This is useful in contexts where `getBoundingClientRect` is not
- * suitable, such as when the element is transformed.
+ * parent, with subpixel precision. Values reflect the real measures before any
+ * potential scaling distortions along the X and Y axes.
  *
- * **Note:** the `left` and `right` values are adjusted due to a limitation
- * in the way the browser calculates the offset position of the element,
- * which can cause unwanted scrollbars to appear. This adjustment makes the
- * values potentially inaccurate within a range of 1 pixel.
+ * Useful in contexts where plain `getBoundingClientRect` calls or `ResizeObserver`
+ * entries are not suitable, such as when the element is transformed, and when
+ * `element.offset<Top|Left|Width|Height>` methods are not precise enough.
  */
 export function getElementOffsetRect(
 	element: HTMLElement
 ): ElementOffsetRect {
+	// Position and dimension values computed with `getBoundingClientRect` have
+	// subpixel precision, but are affected by distortions since they represent
+	// the "real" measures, or in other words, the actual final values as rendered
+	// by the browser.
+	const rect = element.getBoundingClientRect();
+	const offsetParentRect =
+		element.offsetParent?.getBoundingClientRect() ??
```

Comment:
If the offset parent also was affected by a transform compared to its own offset parent, would that have an impact? In other words, do we need recursion going up the DOM tree?

Reply:
This is something I thought of, and it just seems like an edge case we just don't have. Even then, it seems like a relative measure between parent and child would be of little use in a context where the parent and the child have different scaling values applied.